src/main/java/com/thealgorithms/conversions/TemperatureConversion.java

```diff
@@ -0,0 +1,149 @@
+package com.thealgorithms.conversions;
+
+import java.text.DecimalFormat;
+import java.util.Scanner;
+/**
+ * Convets the temperatures from user to Celsius/Fahrenheit/Kelvin
+ *
+ * @author Tharun S.M.
+ */
+
+public class TemperatureConversion {
+    /*
+     * Object of DecimalFormat Class
+     * to round the output to 2 decimal point value
+     */
+    private static final DecimalFormat rnd = new DecimalFormat("0.00");
```

Comment:
Why can't we just convert `double` to `double`? I would suggest that.

Reply:
the slope approach, im not well aware of that subject, and the value is set to float to stop huge decimal value inputs, but the decimal format is required to round the values

Reply:
do u want to use  Affine Converter instead of the conditional if statements?

Reply:
In my opinion, the most natural type to represent temperature is `double`. There is no need to do any rounding.

Regarding, the `AffineConverter`, among the temperature units, which you have suggested, each conversion can be done with a formula of the type
```
valueInNewUnit = someSlope * valueInOldUnit + someIntercept
```
I suggested to have such `AffineConverter`, because then the problem of changing the units can be simplified just to _hard coding_ the slope and the intercept (e.g. in case of converting Kelvin to Celsius, the slope is 1, and the intercept is -273.15).

Reply:
ok ill look into it

Reply:
do you want me to hard code the user inputs also?, because how do we recognize what temperature type is being input by the user